### ISSUE TITLE
Custom headerRight/headerLeft buttons are not clickable on Android

### ISSUE THREAD

**Issue by reinvanimschoot** [react-navigation/react-navigation#7052]
I just noticed that custom buttons on Android are not clickable.

The simplest example would be a one screen stack navigator:

```
const TestStack = createStackNavigator({
  TestScreen: {
    screen: TestScreen,
    navigationOptions: ({navigation}) => ({
      headerRight: (
        <TouchableOpacity
          onPress={() => console.log('Button is Pressed!') }>
          <Text>PRESS ME</Text>
        </TouchableOpacity>
      ),
    }),
  },
```

Everything works flawless on iOS but on Android I get not press feedback.
I tried adding elevation, a different combination of headerModes, but even in this simple example it's not working.

**Comment by gavinangym** [comment]
I'm having this problem on v5 actually. Android only. The JSX inside the TouchableOpacity renders correctly, but logging the output of the onPress shows nothing is being called.

**Comment by elvispap** [comment]
Hey guys...I am having the same issue as well. I also posted on StackOverflow, you can see [here](https://stackoverflow.com/questions/60441961/react-native-onpress-event-doesnt-work-in-android). Let me know if you find out the cause of the issue or I am willing to create a small repo where the issue occurs. @satya164 let me know if you are going to upload the minimal repo which demonstrates the issue.

**Comment by gavinangym** [comment]
https://snack.expo.io/@gavinang/excited-pudding

Repro snack

**Comment by gavinangym** [comment]
I created an expo snack, but that one seems to work. Not sure if its the emulator or what. Still tracing the differences between my actual app implementation and the snack.

**Comment by gavinangym** [comment]
Updating to the following packages seems to have resolved it for me:
    "@react-navigation/stack": "5.1.0", //was 5.0.5
    "@react-navigation/drawer": "5.1.0", //was 5.0.5
    "@react-navigation/native": "5.0.8", //was 5.0.5
    "@react-navigation/bottom-tabs": "5.1.0", //was 5.0.5

**Comment by tamangsuresh** [comment]
Any one have solution for v4.

**Comment by raajnadar** [comment]
upgrade to latest version

**Comment by sadikyalcin** [comment]
I had the same issue on one screen (StackNavigator) on v4. O only have one screen on my stack now so removing `headerMode`, `transitionConfig` and disabling gestures fixed it. @tamangsuresh give it a try.

**Comment by nanacnote** [comment]
Please make sure headerMode="screen" for some reason "float" does not work

**Comment by sabun123** [comment]
I was having this issue with React Navigation v5 as well, my custom headerLeft component was not fully clickable on Android but was totally fine on iOS. Basically, only at the very left tip of the screen on Android would headerLeft start recognizing user presses. I figure something invisible is sitting in the middle of the header overlapping my customer headerLeft.

In the styles of my custom headerLeft component, I just did:
```javascript
  buttonContainer: {
    ...
    zIndex: 100, // fixes Android not allowing full touchableopacity to be press-able
  },
```

Adding a high z-index seems to do the trick, and my custom headerLeft component works as expected now on Android.

This is on the following versions of this package:
```
    "@react-navigation/drawer": "5.9.3",
    "@react-navigation/native": "5.7.4",
    "@react-navigation/stack": "5.9.3",
```

**Comment by aadityapaliwal94** [comment]
I am still facing this issue in Android. I am using react-navigation v5. Tried the above solutions but they didn't work. iOS is working fine.

My stack

`const HomeStack = () => {
  return (
    <Stack.Navigator mode='card' headerMode={"screen"} screenOptions={{
      cardStyleInterpolator: CardStyleInterpolators.forHorizontalIOS
    }}>
      <Stack.Screen name='Home' component={Reader} options={({ route, navigation }) => ({ title: "Comic", headerShown:true, header: () => (<HeaderBar route={route} navigation={navigation} />) })} />
    </Stack.Navigator>
  )
}`

`"@react-navigation/compat": "^5.3.14",
    "@react-navigation/drawer": "^5.12.4",
    "@react-navigation/native": "^5.9.3",
    "@react-navigation/stack": "^5.14.3",`

**Comment by 256hz** [comment]
I doubt this is your problem @aadityapaliwal94, but I was just facing this issue and it was because I had defined conflicting styles in my `headerRight` component and my `headerRightContainerStyle`.  Specifically I had set `position: 'absolute', height: 0, width: 0` in `headerRightContainerStyle`, and then defined the `headerRight` component normally.  I'm surprised the component displayed at all, but I am not surprised that it wasn't clickable.

**Comment by jonorril** [comment]
> I was having this issue with React Navigation v5 as well, my custom headerLeft component was not fully clickable on Android but was totally fine on iOS. Basically, only at the very left tip of the screen on Android would headerLeft start recognizing user presses. I figure something invisible is sitting in the middle of the header overlapping my customer headerLeft.
> 
> In the styles of my custom headerLeft component, I just did:
> 
> ```js
>   buttonContainer: {
>     ...
>     zIndex: 100, // fixes Android not allowing full touchableopacity to be press-able
>   },
> ```
> 
> Adding a high z-index seems to do the trick, and my custom headerLeft component works as expected now on Android.
> 
> This is on the following versions of this package:
> 
> ```
>     "@react-navigation/drawer": "5.9.3",
>     "@react-navigation/native": "5.7.4",
>     "@react-navigation/stack": "5.9.3",
> ```

This works. 😊
Thanks @sabun123

**Comment by eddievlagea** [comment]
@zvs001 have you found a fix?

**Comment by zvs001** [comment]
Yes.
it happens for me when I use  `headerTransparent: true`

I solved it by adding `headerMode` param:
```typescript
screenOptions: {
    headerMode: 'float',
}
```

**Comment by akinyele** [comment]
Was having this issue earlier. Just incase this helps anyone, what solved it for me was removing `position: 'absolute'`.

**Comment by hernanspc** [comment]
sasa

**Comment by shaquillehinds** [comment]
I had the same issue but for me it was because I had set tabBarStyle: {display: 'none'} in my bottomTabNavigator screenOptions. I'm using react navigation 6 so the only way to hide the bottom nav is to set the display to none or translateY. Using translateY to hide the bottom nav instead of display fixed it for me

**Comment by windyinwind** [comment]
The above solutions are not working for me.  It turned out that I was importing TouchableOpacity from 'react-native-gesture-handler' 
After changing to import TouchableOpacity from 'react-native'.  it started working for me.

**Comment by kriit24** [comment]
Same thing in IOS [https://github.com/expo/expo/issues/32927](https://github.com/expo/expo/issues/32927)

**Comment by abikali** [comment]
I'm still facing this issue.

```
import React from 'react';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { HomeStackParamList } from '../../types/routes';
import { HomeHeaderLeft, HomeHeaderRight } from '@/src/features/home/components/Header';
import { theme } from '@/src/app/theme';
import { HomeScreen } from '@/src/features/home/screens/HomeScreen';
import { NotificationsScreen } from '@/src/features/notifications/screens/NotificationsScreen';
import { TouchableOpacity, Text, Alert, Button, View } from 'react-native';

const Stack = createNativeStackNavigator<HomeStackParamList>();

const DebugHeaderLeft = () => {
  const handlePress = () => {
    Alert.alert('Pressed', 'Debug header was pressed');
  };

  return (
    <View style={{ marginLeft: 10, zIndex: 100 }}>
      <Button
        onPress={handlePress}
        title="Test"
        color="lightblue"
      />
    </View>
  );
};

export const HomeStackNavigator = () => {
  return (
    <Stack.Navigator
      screenOptions={{
        headerShown: true,
        headerTitleAlign: 'center',
        headerTitleStyle: { fontFamily: theme.fontFamily.bold },
      }}
    >
      <Stack.Screen
        name='home'
        component={HomeScreen}
        options={{
          headerTitle: '',
          headerLeft: () => <DebugHeaderLeft />,
          headerRight: () => <HomeHeaderRight />,
        }}
      />
      <Stack.Screen
        name='notifications'
        component={NotificationsScreen}
        options={{
          headerTitle: 'Notifications',
          headerBackVisible: true,
          headerTintColor: theme.colors.black
        }}
      />
    </Stack.Navigator>
  );
};
```

**Comment by guillaumehcht** [comment]
I'm experiencing the same issue as well. The `Pressable` in the `headerRight` works perfectly fine on iOS and doesn't work at all on Android. Anyone has a solution? 

I'm on RN0.76 and Expo52

**Comment by rajgopalv** [comment]
Same issue in my android device (Samsung s10) - works perfect in the simulator. (pixel 8 pro )

**Comment by TierryBr** [comment]
Same problem using RN 0.76 and latest version of the lib, if you use onPressIn as @guillaumehcht mentioned it works

**Comment by insaindesign** [comment]
Using `onPressIn` instead of `onPress` prop works, you can also nest it in a `<View />` component.

e.g., `headerRight: (props: { tintColor: string }) => <View><Button onPress={() => console.log("pressed")} /></View>`.

I suspect the prop is being overwritten or swallowed somewhere.

**Comment by RohovDmytro** [comment]
Same. onPressIn works though.

**Comment by JcbPrn** [comment]
Still same issue here. Before I had the issue for all the `TouchableOpacity` components on iPhone XS and SE. After updating to newest version of `react-native-screens` and `react-navigation` I only have the issue in the headerRight on iPhone XS and SE. `onPressIn` and `onPressOut` are also working for me.

**Comment by HyunJinNo** [comment]
I have the same issue. `onPress` doesn't work, but `onPressIn` and `onPressOut` are working properly.

```typescript
export const DiaryEditorScreen = () => {
  ...

  useEffect(() => {
    navigation.setOptions({
      // eslint-disable-next-line react/no-unstable-nested-components
      headerRight: () => <DiaryRegisterButton methods={methods} />,
    });
  }, [methods, navigation]);

  ...
}
```

```typescript
import {tw} from '@src/libs/tailwind';
import {Diary} from '@src/types/diary';
import React, {useState} from 'react';
import {UseFormReturn} from 'react-hook-form';
import {Pressable, Text} from 'react-native';

interface DiaryRegisterButtonProps {
  methods: UseFormReturn<Diary, any, undefined>;
}

export const DiaryRegisterButton = ({methods}: DiaryRegisterButtonProps) => {
  const [count, setCount] = useState(0);
  const handleSubmit = () => {
    setCount(value => value + 1);
    console.log(methods.getValues('title'));
  };

  return (
    <Pressable
      style={({pressed}) => {
        return tw.style([pressed ? 'bg-slate-100' : '', 'rounded-2xl p-2']);
      }}
      onPressOut={() => handleSubmit()}>
      <Text style={tw`font-semibold text-primary-green`}>{count}</Text>
    </Pressable>
  );
};
```

**Comment by dineshpithiya** [comment]
In my case, I encountered the same issue, but it doesn't occur on all devices. I found this issue specifically on the Samsung Galaxy S22. The reason behind it is as follows:

**Not working code:**
```
<TouchableOpacity>
  <Feather onPress={() => router.replace('groups')} style={{color: "white"}} name={backIcon} size={25} />
</TouchableOpacity>
```

**Working code:**
```
<TouchableOpacity onPress={() => router.replace('groups')}>
  <Feather style={{color: "white"}} name={backIcon} size={25} />
</TouchableOpacity>
```

Also you can try replace `onPress` to `onPressOut` or `onPressIn` as well

**Comment by RohovDmytro** [comment]
@dineshpithiya were you using new arch while experiencing the issue?

**Comment by dineshpithiya** [comment]
@RohovDmytro I using below versions 

 "dependencies": {
    "@expo/vector-icons": "^14.0.2",
    "@react-native-async-storage/async-storage": "^2.1.0",
    "@react-navigation/bottom-tabs": "^7.0.0",
    "@react-navigation/drawer": "^7.0.11",
    "@react-navigation/native": "^7.0.0",
    "@rneui/themed": "^4.0.0-rc.8",
    "axios": "^1.7.9",
    "buffer": "^6.0.3",
    "crypto-es": "^2.1.0",
    "expo": "~52.0.11",
    "expo-blur": "~14.0.1",
    "expo-clipboard": "~7.0.0",
    "expo-constants": "~17.0.3",
    "expo-dev-client": "~5.0.9",
    "expo-font": "~13.0.1",
    "expo-haptics": "~14.0.0",
    "expo-image-picker": "^16.0.3",
    "expo-linking": "~7.0.3",
    "expo-localization": "~16.0.0",
    "expo-router": "~4.0.9",
    "expo-secure-store": "^14.0.0",
    "expo-splash-screen": "~0.29.13",
    "expo-status-bar": "~2.0.0",
    "expo-symbols": "~0.2.0",
    "expo-system-ui": "~4.0.4",
    "expo-updates": "~0.26.9",
    "expo-web-browser": "~14.0.1",
    "i18next": "^24.2.0",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "react-i18next": "^15.4.0",
    "react-native": "0.76.3",
    "react-native-autogrow-textinput": "^5.4.0",
    "react-native-chart-kit": "^6.12.0",
    "react-native-element-dropdown": "^2.12.2",
    "react-native-flexbox-grid": "^0.3.2",
    "react-native-gesture-handler": "~2.20.2",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-modal": "^13.0.1",
    "react-native-otp-entry": "^1.7.3",
    "react-native-ratings": "^8.1.0",
    "react-native-reanimated": "~3.16.1",
    "react-native-safe-area-context": "4.12.0",
    "react-native-screens": "~4.1.0",
    "react-native-view-more-text": "^2.2.0",
    "react-native-web": "~0.19.13",
    "react-native-webview": "13.12.2"
  },

**Comment by dineshpithiya** [comment]
I'm facing an issue with the accordion in my drawer list. It seems that only the `onPress` event is supported, but it's not working as expected after scroll. Here's the relevant code I'm using:

` **onPress={() => accordionToggle(index)}>**`

```
 const accordionToggle = (index) => {
    setExpandedIndex(expandedIndex === index ? null : index);
  };

<DrawerContentScrollView contentContainerStyle={{ paddingTop: 0 }} {...props} style={{padding:0}}>
{drawerItems.map((item, index) => (
          <View key={item.key} style={{ padding:0 }}>
            <ListItem.Accordion
              key={item.key}
              icon={<Feather name={"chevron-down"} size={24} color={"grey"}/>}
              containerStyle={styles.accordionHeaderContainer}
              content={
                <>
                  <Feather name={item.icon} size={24} containerStyle={styles.navIcon} color={"grey"} />
                  <ListItem.Content>
                    <ListItem.Title style={[styles.navItemLabel,{paddingLeft:5}]}>{item.label}</ListItem.Title>
                  </ListItem.Content>
                </>
              }
              isExpanded={expandedIndex === index}
              **onPress={() => accordionToggle(index)}>**
              {item.links.length>0 && (  
              item.links.map((subItem, subIndex) => (
              <TouchableOpacity key={subItem.key} onPressOut={subItem.onPress}>
                <ListItem bottomDivider>
                  <ListItem.Content>
                    <ListItem.Title style={[styles.navItemLabel]}>{subItem.name}</ListItem.Title>
                  </ListItem.Content>
                  <ListItem.Chevron size={24}/>
                </ListItem>
              </TouchableOpacity>
              )))}
            </ListItem.Accordion>
          </View>
)}
</DrawerContentScrollView>
```

**Comment by Martijncvv** [comment]
Was using the HeaderBackButton from react-navigation elements and didn't work anymore either on android after upgrading to Expo 52 (It did work on web).
Changed it for a custom `Pressable` component with `onPressIn` and it works now; Just `onPress` doesn't. 
Thanks :)

**Comment by pinguluk** [comment]
I have the same problem. I've tried with ```onPressOut```, which triggers the event, but it actually behaves as ```onPressIn```, when using  in the header component (I share the same component both in header and another location and it behaves normally at the other location)

**Comment by swiftdevil** [comment]
I'm facing a very similar issue and am debugging it. For whatever reason, the `UIManager.measure()` function is passing in some weird values to the `Pressability._measureCallback()` specifically for the Pressable that's rendered in the navigation header.

Measurement values for the header Pressable:
![Image](https://github.com/user-attachments/assets/7f291ab6-8e90-4212-ad96-51df3b49eea7)

Measurement values for a non-header Pressable:
![Image](https://github.com/user-attachments/assets/7db555c1-3801-4bfa-907c-5c10002d6457)

This is where these bad values are causing the problem... in `Pressability._isTouchWithinResponderRegion()`:
![Image](https://github.com/user-attachments/assets/12930b23-8445-4fe8-9fb6-22f86f023593)
It thinks the press occurs outside of the Responder, even though the responder is on the screen and can be interacted with.


Notice the `pageY` values in these screenshots. The header Pressable is measured at `pageY=-537`, which is way off-screen. If I had to guess, I think this is due to the Pressable being rendered outside of the "dom" (not sure what it's called in RN) and therefore gets assigned invalid values... could be wrong though.


The header button is set up as follows:
```jsx
useEffect(() => {
    navigation.setOptions({
        headerRight: () =>
            <IconButton 
                icon={'check-all'}
                //pressRetentionOffset={1000}
                onPress={() => setCompleteVisible(!isCompleteVisible)}
            />
    });
}, []);
```

The commented line above is the workaround I'm using (ugly, but gets the job done for now).

**Comment by abedolinger** [comment]
@swiftdevil interesting find and nice debugging! It might be related to [this issue in Reanimated's `measure`](https://github.com/software-mansion/react-native-reanimated/issues/3188) which may still be unresolved.  If it is, it may be worth posting your findings in that thread - a maintainer could be interested if it could solve this as well.

**Comment by swiftdevil** [comment]
Thanks @abedolinger! I was able to track this further... the fix for the problem I was having is here: https://github.com/software-mansion/react-native-screens/pull/2466 and was released in `react-native-screens` v4.6.0. Upgrading to the fixed version solved it for me (I didn't have to update any other dependency versions either, which is quite nice).

**Comment by sbigstar0310** [comment]
> In my case, I encountered the same issue, but it doesn't occur on all devices. I found this issue specifically on the Samsung Galaxy S22. The reason behind it is as follows:
> 
> **Not working code:**
> 
> ```
> <TouchableOpacity>
>   <Feather onPress={() => router.replace('groups')} style={{color: "white"}} name={backIcon} size={25} />
> </TouchableOpacity>
> ```
> 
> **Working code:**
> 
> ```
> <TouchableOpacity onPress={() => router.replace('groups')}>
>   <Feather style={{color: "white"}} name={backIcon} size={25} />
> </TouchableOpacity>
> ```
> 
> Also you can try replace `onPress` to `onPressOut` or `onPressIn` as well

Thank you so much! I have solved by using `onPressOut` on Galaxy S22
